src/include/OpenImageIO/dassert.h

```diff
@@ -92,15 +96,10 @@
 #endif
 
 
-/// DASSERT(condition) is just like ASSERT, except that it only is
-/// functional in DEBUG mode, but does nothing when in a non-DEBUG
-/// (optimized, shipping) build.
-#ifndef NDEBUG
-#    define DASSERT(x) ASSERT(x)
-#else
-/* DASSERT does nothing when not debugging; sizeof trick prevents warnings */
-#    define DASSERT(x) ((void)sizeof(x)) /*NOLINT*/
-#endif
+/// DASSERT(condition) is just an alias for the usual assert() macro.
+/// It does nothing when in a non-DEBUG (optimized, shipping) build.
+#define DASSERT(x) assert(x)
```

Comment:
The `sizeof` trick was useful in preventing warnings about unused variables in some cases. Most assert implementations (sadly) don't implement this.

Maybe it's worth keeping the `#ifdef NDEBUG` case ?

Reply:
So the main point of all this was to want to find a way that a function using DASSERT can be decorated with `__host__ __device__` without generating warnings.

My ASSERT when used in Cuda code will generate warnings upon compile (because of its referencing of stderr), and not work properly when executed. Whereas if you use assert(), there is a Cuda implementation that not only works but prints some useful info on the threads, etc. It uses some trickery of substituting a different assert.h or some other override of a true assert() function that is Cuda-aware.

So what I was trying to do is make DASSERT just use the built in assert(), and only for my unconditional ASSERT have our own implementation.

I guess I can go back to the original and have an even more complicated `#if` that falls back to assert() when in NVCC only, or something like that?

Reply:
OH, I see your point about the `#if`. Let me give it another try...